### ISSUE TITLE
When hiding all fields in a fieldset, the fieldset remains, but is empty. it should be hidden too.

### ISSUE THREAD

**Issue by plazmus** [backdrop-contrib/conditional_fields#23]
If all the fields in a fieldset are hidden, then the fieldset itself should be hidden rather than displaying an empty fieldset.

This is a working patch taken from https://www.drupal.org/project/conditional_fields/issues/1903226

``` patch
---
 conditional_fields.js | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/conditional_fields.js b/conditional_fields.js
index 148f793..d16eca7 100644
--- a/conditional_fields.js
+++ b/conditional_fields.js
@@ -141,4 +141,24 @@ Backdrop.behaviors.conditionalFields = {
   }
 };
 
+// Hide fieldsets unless one of their fields is visible.
+  $(document).bind('state:visible', function(e) {
+    if (e.trigger) {
+      var $wrapper = $(e.target).closest('.fieldset-wrapper');
+      if ($wrapper.length) {
+        var hidden = true;
+        $wrapper.children(':not(.fieldset-description)').each(function (index, element) {
+          if ($(element).css('display') != 'none') {
+            hidden = false;
+          }
+        });
+        if (hidden) {
+          $wrapper.closest('fieldset').hide();
+        } else {
+          $wrapper.closest('fieldset').show();
+        }
+      }
+    }
+  });
+
 })(jQuery);

```

**Comment by laryn** [comment]
@plazmus Is this for fieldsets that are added by the Field Group module? (Side note: that module is being advocated to be added to core by @jenlampton and has [work being done here](https://github.com/backdrop/backdrop-issues/issues/647)).

If so, have you tested with other group types, such as Details elements? It would be nice to cover all those bases if possible.

**Comment by plazmus** [comment]
Yes, Field Group module. No, I haven't tested with other group types.

**Comment by yorkshire-pudding** [comment]
@laryn I'm interested in this ability. Does it need to be an all or nothing approach? Would it be possible to get compatibility with field groups in based on this with a ticket to get other types in the future?

I've tested the patch with field groups and works well.

**Comment by aitala** [comment]
I have run into this as well...

And I'm not sure the proposed patch actually works.

Eric

**Comment by laryn** [comment]
@yorkshire-pudding Sorry for the late reply. It doesn't necessarily need to be all or nothing, but I think field groups and details elements should be covered. There seems to be some question about whether this patch works for field groups in its current state, and I see some deprecated syntax (e.g. `.bind()` at least) so it needs some improvement.

**Comment by olafgrabienski** [comment]
I'm also interested in hiding an empty field group. I've tested the patch, and it works somehow for me. It didn't work rightaway but only after switching the jQuery version from 3.x to 1.x and back. My steps:

- Create a node where a condition can lead to an empty fieldset in the node edit form.
- Notice there is an empty fieldset visible.
- Apply the patch.
- Edit the node. The empty fieldset is still visible.
- Flush CSS and JS cache. No changes.
- Flush all cashes. No changes.
- Switch jQuery to version 1.x. The empty fieldset isn't visible anymore. 👍🏼
- Switch jQuery back to version 3.x. The empty fieldset remains invisible. 👍🏼
- Clear caches. Same as before. 👍🏼
- Change browser, log in, and edit the node there. Same as before. 👍🏼

@yorkshire-pudding and @aitala Does the patch work for you as well, when you follow those steps?

> I see some deprecated syntax (e.g. `.bind()` at least)

Can't help with this, but I'm happy to test further.

**Comment by olafgrabienski** [comment]
After updating Conditional Fields to [release 1.x-3.0.2](https://github.com/backdrop-contrib/conditional_fields/releases/tag/1.x-3.0.2), I made sure that the issue persists. I've then added the code from the patch to the current version, and it still fixes the issue for me (this time even without switching jQuery versions forth and back (see last comment)).

@yorkshire-pudding – Are you maybe interested to provide a PR which includes also a fix for the details element?

@laryn – regarding deprecated syntax, `.bind()` is all over the place in the file, shouldn't that be a separate issue?
